### DISCUSSION TITLE
Filter the jobs based on paths

### DISCUSSION THREAD

**Discussion by cilerler** [community/community#26177] [Actions]
<p>I would like to be able to filter a job as we do on the action with <code>**on.push.paths**</code></p>
<p>Any idea how to do that, or should this be a feature request?</p>
<p>Ideally, it behaves the same way as it does on <code>on.push.paths</code>. But I would be fine to having an environment variable (something like **<code>github.push_paths</code>) **that holds all the paths and having a way to compare it against that.</p>
<p>I found these articles (<a href="https://github.community/t/find-what-files-changed-in-a-pushed-commit/17037">article1</a>, <a href="https://github.community/t/check-pushed-file-changes-with-git-diff-tree-in-github-actions/17220">article2</a>), but neither of those is solves what I would like to do.</p>
<p><strong>A sample having paths under the jobs</strong></p>
<pre><code>name: Ruya

on:
  push:
    branches:
    - master
    paths:
    - 'src/Ruya.Primitives/**'
    - 'src/Ruya.Helpers.Primitives/**'
    - 'src/Ruya.AppDomain/**'

jobs:

  build:
    if: contains(toJson(github.event.commits), '[skip ci]') == false
    runs-on: ubuntu-latest
    steps:
      ...

  primitives:
    paths:
- 'src/Ruya.Primitives/**'
    runs-on: ubuntu-latest
    steps:
      ...

  primitives-helper:
    needs: primitives
    paths:
- 'src/Ruya.Primitives/**'
- 'src/Ruya.Helpers.Primitives/**'
    runs-on: ubuntu-latest
    steps:
      ...

  app-domain:
    paths:
- 'src/Ruya.AppDomain/**'
    runs-on: ubuntu-latest
    steps:
      ...
</code></pre>
<p><strong>A sample having paths under the environment variable and assuming that <code>contains</code> is supporting globe patterns</strong></p>
<pre><code>name: Ruya

on:
  push:
    branches:
    - master
    paths:
    - 'src/Ruya.Primitives/**'
- 'src/Ruya.Helpers.Primitives/**'
- 'src/Ruya.AppDomain/**'

jobs:

  build:
    if: contains(toJson(github.event.commits), '[skip ci]') == false
runs-on: ubuntu-latest
steps:
...

primitives:
if:contains(toJson(github.push_paths), 'src/Ruya.Primitives/**') == true
runs-on: ubuntu-latest
steps:
...

primitives-helper:
needs: primitives
if:contains(toJson(github.push_paths), 'src/Ruya.Primitives/**') == true ||contains(toJson(github.push_paths), 'src/Ruya.Helpers.Primitives/**') == true
runs-on: ubuntu-latest
steps:
...

app-domain:
if:contains(toJson(github.push_paths), 'src/Ruya.AppDomain/**') == true
runs-on: ubuntu-latest
steps:
...
</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/cilerler">@cilerler</a> ,</p>
<p>Currently, in the <strong>commits</strong> object of the <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a>, there indeed is not a <strong>property</strong> (like <strong>push_paths</strong> as you mentioned) to list the changed files or paths on the push.</p>
<p>Using the <a href="https://git-scm.com/docs/git-diff" rel="nofollow noopener"><strong>git-diff</strong></a> command indeed can get the list of the changed files. I tried to set the list as an environment variable (or an output parameter) to be used in the if conditional for the jobs, but I failed, and this way is quite troublesome and unreliable.</p>
<p>I’m afraid that currently there is not any available and easy method to do this.</p>
<p>If your projects really need this feature, I recommend you directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>
